flexmeasures/data/models/reporting/__init__.py

```diff
@@ -48,7 +44,13 @@ def __init__(
 
         self.reporter_config_raw = reporter_config_raw
 
-    def fetch_data(self):
+    def fetch_data(
+        self,
+        start: datetime,
+        end: datetime,
+        input_resolution: timedelta = None,
```

Comment:
For the type annotation: `timedelta | None` or `Optional[timedelta]` (I prefer the former, more modern syntax, and have started moving our codebase towards this syntax for some time. You'd need to add a `from __future__ import annotations` so that we keep supporting Python 3.9)

```
input_resolution: timedelta | None = None,
```